src/main/java/no/nav/pto/veilarbportefolje/persononinfo/barnUnder18Aar/BarnUnder18AarService.java

```diff
@@ -50,24 +50,31 @@ public List<Fnr> hentBarnFnrsForForeldre(List<Fnr> fnrForeldre) {
     }
 
     public void lagreBarnOgForeldreansvar(Fnr foresattIdent, List<BarnUnder18Aar> barnFraPdl) {
-        List<Fnr> lagredeBarn = barnUnder18AarRepository.hentForeldreansvarForPerson(foresattIdent);
+        try {
+            List<Fnr> lagredeBarn = barnUnder18AarRepository.hentForeldreansvarForPerson(foresattIdent);
 
-        List<Fnr> barnFnrFraPdl = barnFraPdl.stream().map(BarnUnder18Aar::getFnr).toList();
+            List<Fnr> barnFnrFraPdl = barnFraPdl.stream().map(BarnUnder18Aar::getFnr).toList();
 
-        lagredeBarn.forEach(barnFnr -> {
-            if (!barnFnrFraPdl.contains(barnFnr)) {
-                barnUnder18AarRepository.slettForeldreansvar(foresattIdent, barnFnr);
-                slettBarnDataHvisIngenForeldreErUnderOppfolging(barnFnr);
-                secureLog.warn(String.format("Barn fjernet fra PDL for foreldre %s og barn %s", foresattIdent, barnFnr));
-            }
-        });
-
-        barnFraPdl.forEach(barnUnder18Aar -> {
-            if (erUnder18Aar(barnUnder18Aar.getFodselsdato())) {
-                barnUnder18AarRepository.lagreBarnData(barnUnder18Aar.getFnr(), barnUnder18Aar.getFodselsdato(), barnUnder18Aar.getDiskresjonskode());
-                barnUnder18AarRepository.lagreForeldreansvar(foresattIdent, barnUnder18Aar.getFnr());
-            }
-        });
+            lagredeBarn.forEach(barnFnr -> {
+                if (!barnFnrFraPdl.contains(barnFnr)) {
+                    barnUnder18AarRepository.slettForeldreansvar(foresattIdent, barnFnr);
+                    slettBarnDataHvisIngenForeldreErUnderOppfolging(barnFnr);
+                    secureLog.warn(String.format("Barn fjernet fra PDL for foreldre %s og barn %s", foresattIdent, barnFnr));
+                }
+            });
+
+            barnFraPdl.forEach(barnUnder18Aar -> {
+                if (erUnder18Aar(barnUnder18Aar.getFodselsdato())) {
+                    barnUnder18AarRepository.lagreBarnData(barnUnder18Aar.getFnr(), barnUnder18Aar.getFodselsdato(), barnUnder18Aar.getDiskresjonskode());
+                    barnUnder18AarRepository.lagreForeldreansvar(foresattIdent, barnUnder18Aar.getFnr());
+                }else{
+                    secureLog.debug("Barn is older then 17 yrs: " + barnUnder18Aar.getFodselsdato());
+                }
+            });
+        }
+        catch (Exception e){
+            secureLog.error("Kan ikke lagre data om barn og foreldreansvar for person: " + foresattIdent + ". Antall av barn: " + barnFraPdl.size());
```

Comment:
I am just a bit curious about this sentence, because I think it could catch an error but still have been able to do parts of the saving....? If it fails on the 3d child for example? Or I am wrong?

Reply:
Yep, that is true...we need to discuss a bit what is solution..